test/Integration/tpp-run-dequantize-quantize-correctness-i8-f32-i8.mlir

```diff
@@ -0,0 +1,164 @@
+    %neg_exp2 = tensor.empty() : !oneDimf32
+	  %res_exp_val = linalg.generic {
+      indexing_maps = [#map0, #map0],
+      iterator_types = ["parallel"]
+    } ins(%absmax : !oneDimf32) outs(%res_exp : tensor<3xi32>) {
+	    ^bb0(%in: f32, %out: i32):
+      %fr_exp = llvm.intr.frexp(%in) : (f32) -> !llvm.struct<(f32, i32)>
+      %a = llvm.extractvalue %fr_exp[1] : !llvm.struct<(f32, i32)>
+      %b = llvm.extractvalue %fr_exp[0] : !llvm.struct<(f32, i32)>
+      %unbiased_exp = arith.subi %a, %c7 : i32
+
+      linalg.yield %unbiased_exp : i32
+	  } -> tensor<3xi32>
+
+    // Compute 2 raise to the power of -ve value of the exponent
+    %neg_exp2_val = linalg.generic {
+      indexing_maps = [#map0, #map0],
+      iterator_types = ["parallel"]
+    } ins(%res_exp_val : tensor<3xi32>) outs(%neg_exp2 : !oneDimf32) {
+      ^bb0(%in: i32, %out: f32):
+        %neg = arith.subi %cst0, %in : i32
+        %castToFloat = arith.sitofp %neg : i32 to f32
+        %n_exp2 = math.exp2 %castToFloat : f32
+        linalg.yield %n_exp2 : f32
+    } -> !oneDimf32
+
+    %pos_exp2 = tensor.empty() : !oneDimf32
+    %pos_exp2_val = linalg.generic {
+      indexing_maps = [#map0, #map0],
+      iterator_types = ["parallel"]
+    } ins(%res_exp_val : tensor<3xi32>) outs(%pos_exp2 : !oneDimf32) {
+      ^bb0(%in: i32, %out: f32):
+        %castToFloat = arith.sitofp %in : i32 to f32
+        %p_exp2 = math.exp2 %castToFloat : f32
+        linalg.yield %p_exp2 : f32
+    } -> !oneDimf32
+
+    return %neg_exp2_val, %pos_exp2_val : !oneDimf32, !oneDimf32
+  }
+
+  // ========================== Quantize =================================
+  func.func @quantize(%dquant_val: !twoDimf32) -> (!twoDimi8, !oneDimf32) {
+    %c0_f32 = arith.constant 0.0 : f32
+    %c0 = arith.constant 0 : index
+    %c3 = arith.constant 3 : index
+    %c1 = arith.constant 1 : index
+
+    %neg_exp2_val, %pos_exp2_val = call @computeScalingFactors(%dquant_val) : (!twoDimf32) -> (!oneDimf32, !oneDimf32)
+
+    // Broadcast and Scale Cf32
+    %scaleRes = tensor.empty() : !twoDimf32
+    %bcastScaleRes = linalg.broadcast ins(%neg_exp2_val:!oneDimf32) outs(%scaleRes:!twoDimf32) dimensions = [1]
+
+    %scaledC = tensor.empty() : !twoDimf32
+    %scaleRes_val = linalg.mul ins(%dquant_val, %bcastScaleRes : !twoDimf32, !twoDimf32) outs(%scaledC : !twoDimf32) -> !twoDimf32
+    
+    // Down-convert Cf32 -> Ci8
+    %castToi8 = tensor.empty() : !twoDimi8
+    %castToi8_val = linalg.generic {
+      indexing_maps = [#map1, #map1],
+      iterator_types = ["parallel", "parallel"]
+    } ins(%scaleRes_val : !twoDimf32) outs(%castToi8 : !twoDimi8) {
+      ^bb0(%in_val: f32, %out_val: i8):
+        %clamped_i8 = arith.fptosi %in_val : f32 to i8
+        linalg.yield %clamped_i8 : i8
+    } -> !twoDimi8
+
+    return %castToi8_val, %pos_exp2_val : !twoDimi8, !oneDimf32
+  }
+
+  func.func @entry(%input : !twoDimi8, %scale : !oneDimf32, %output : !twoDimi8) -> (!twoDimi8) {
```

Comment:
I imagine the `%scale` is produced by the int init via the float init, but there's no way I can generically guess which argument is the scaling factor. Is this just a coincidence that the one you're initializing has the same shape as the one you created from inside the integer init code?

What if you have two quantized arguments (and two scaling factors)? Will they just use the same global tensor?